### PR TITLE
feat(webex): Add Webex integration

### PR DESCRIPTION
**Pull request by damienalexandre**
Refs: #272

# To be done

- [x] Test more, on a variety of usecases
- [x] Try to get the avatars when present
- [x] Add a logo on the homepage + button
- [x] Improve user feedback for non admin users (they cannot run the app)
- [x] Add pagination support 
- [x] Use groups / teams, see external task: #274

### CODE REVIEW

File: src/Controller/WebexController.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            ];
+            $authUrl = $provider->getAuthorizationUrl($options);
+
+            $session->set(WebexApplication::SESSION_KEY_STATE, $provider->getState());
+
+            return new RedirectResponse($authUrl);
+        }
+
+        // Check given state against previously stored one to mitigate CSRF attack
+        if (empty($request->query->get('state')) || ($request->query->get('state') !== $session->get(WebexApplication::SESSION_KEY_STATE))) {
```
Comment (lyrixx) [inline src/Controller/WebexController.php:74]: 🙏 please, [**never**](https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/) use empty

BTW, the following line does the same job and it simpler


```suggestion
        if ($request->query->get('state') !== $session->get(WebexApplication::SESSION_KEY_STATE) {
```

File: src/Controller/WebexController.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        try {
+            // Try to get an access token (using the authorization code grant)
+            /** @var AccessToken $token */
+            $token = $provider->getAccessToken('authorization_code', [
+                'code' => $request->query->get('code'),
+            ]);
+
+            $appToken = new ApplicationToken($token->getToken());
+            $admin = $userExtractor->getMe($token->getToken());
+        } catch (\Exception $e) {
```
Comment (lyrixx) [inline src/Controller/WebexController.php:88]: Please, never use `\Exception`. It makes debuging much harder. Could you narrow the exception class?

File: src/Webex/MessageSender.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+You have been selected to be part of a Secret Santa 🎅!
+
+Someone will get you a gift and **you have been chosen to gift:**
+
+🎁 **%s** 🎁',
+            $receiverUser->getName()
+        );
+
+        if (!empty($userNote = $secretSanta->getUserNote($receiver))) {
```
Comment (lyrixx) [inline src/Webex/MessageSender.php:45]: 😭 - same everywhere

File: templates/santa/application/participants_webex.html.twig [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            data-search-index="{{ user.name }}"
+    >
+
+        <div class="data-summary"
+             data-summary="{{ current.userSummary(userForm, user)|escape('html_attr')|raw }}">
+            {{ form_widget(userForm, {'attr': {
+                'data-identifier': user.identifier
+            }}
+            ) }}
+        </div>
```
Comment (lyrixx) [inline templates/santa/application/participants_webex.html.twig:30]:
```suggestion
        <div class="data-summary"
             data-summary="{{ current.userSummary(userForm, user)|escape('html_attr')|raw }}"
        >
            {{ form_widget(userForm, { attr: {
                    'data-identifier': user.identifier
                }}
            ) }}
        </div>
```

File: src/Controller/WebexController.php (hunk 2 above)
Comment (damienalexandre) [reply src/Controller/WebexController.php:88]: We call an external library method and which is badly documented, in our case AuthenticationException is catched by a global listener and display a nice page to the user: we did it on purpose.

Comment (lyrixx) [reply src/Controller/WebexController.php:88]: So if a network error occurs (for example), we display a wrong message to the end user?

**Review by pyrech — APPROVED** [review]

**Pull request merged**